lib/fdotypes.c

```diff
@@ -858,6 +858,37 @@ char *fdo_nonce_to_string(uint8_t *n, char *buf, int buf_sz)
 	return buf;
 }
 
+// -----------------------------------------------------------------------------
+// GUID routines
+//
+/**
+ * convert to GUID to string
+ * @param g - pointer to the byte array that holds the GUID
+ * @param buf - pointer to the converted string
+ * @param buf_sz - size of the converted string
+ * @return pointer to the converted string
+ */
```

Comment:
It would be better to put a NULL pointer check for g, g->bytes and buf. This API can be used at some other places too in future considering it is very generic. Having those checks would make it robust.

Reply:
Updated